test/components/form-attachment/list.spec.js

```diff
@@ -1037,15 +1037,31 @@ describe('FormAttachmentList', () => {
   });
 
   describe('dataset linking', () => {
-    beforeEach(() => {
+    it('sends a request for datasets if the project has one', () => {
+      testData.extendedProjects.createPast(1, { forms: 2, datasets: 1 });
+      testData.extendedDatasets.createPast(1);
       testData.extendedForms.createPast(1, { draft: true });
+      testData.standardFormAttachments.createPast(1);
+      return load('/projects/1/forms/f/draft/attachments')
+        .respondWithData(() => testData.extendedDatasets.sorted())
+        .testRequests([
+          null, // project
+          null, // form
+          null, // formDraft
+          null, // attachments
+          { url: '/v1/projects/1/datasets' }
+        ]);
     });
 
-    it('autolinks dataset', async () => {
+    it('renders correctly for an attachment linked to a dataset', async () => {
+      testData.extendedProjects.createPast(1, { forms: 2, datasets: 1 });
+      testData.extendedDatasets.createPast(1, { name: 'shovels' });
+      testData.extendedForms.createPast(1, { draft: true });
       testData.standardFormAttachments.createPast(1, { type: 'file', name: 'shovels.csv', datasetExists: true });
       const component = await load('/projects/1/forms/f/draft/attachments', {
         root: false
-      });
+      })
+        .respondWithData(() => testData.extendedDatasets.sorted());
```

Comment:
This test makes assertions about an attachment that is linked to an entity list. However, for an attachment to be linked to an entity list, there has to be an entity list, so I created one in `testData` and returned it to `load()`. Those changes don't affect whether the test passes or not, but the test now more closely matches how the page works in practice.